File: .agents/skills/create-surface-typescript/templates/surface_name.ts
import { <VO> } from '../shared/<domain>/taxonomy_<name>_vo';
import { I<Name>Aggregate } from '../shared/<domain>/contract_<name>_aggregate';

export class Surface<Name> {
    constructor(private readonly aggregate: I<Name>Aggregate) {}

    handle(event: TuiEvent): Result<UiState, SurfaceError> {
        // orchestration only
        return Ok(UiState.idle());
    }
}

File: .agents/skills/create-surface-python/templates/surface_name.py
from shared.<domain>.taxonomy_<name>_vo import <VO>
from shared.<domain>.contract_<name>_aggregate import I<Name>Aggregate


class Surface<Name>:
    def __init__(self, aggregate: I<Name>Aggregate):
        self._aggregate = aggregate

    def handle(self, event: TuiEvent) -> Result[UiState, SurfaceError]:
        # orchestration only
        return Ok(UiState.idle())
